edx_proctoring/api.py

```diff
@@ -1246,28 +1242,29 @@ def reset_practice_exam(exam_id, user_id):
         )
         raise ProctoredExamIllegalStatusTransition(msg)
 
-    # prevent a reset if the exam is currently in progress
+    # prevent a reset if the exam is currently in progress or has already been verified
     attempt_in_progress = ProctoredExamStudentAttemptStatus.is_incomplete_status(exam_attempt_obj.status)
-    if attempt_in_progress:
+    if attempt_in_progress or exam_attempt_obj.status == ProctoredExamStudentAttemptStatus.verified:
         msg = (
             'Failed to reset attempt status on exam_id {exam_id} for user_id {user_id}. '
-            'Attempt with status {status} is still in progress!'.format(
+            'Attempt with status {status} cannot be reset!'.format(
                 exam_id=exam_id,
                 user_id=user_id,
                 status=exam_attempt_obj.status,
             )
         )
         raise ProctoredExamIllegalStatusTransition(msg)
 
-    exam_attempt_obj.status = ProctoredExamStudentAttemptStatus.created
-    exam_attempt_obj.started_at = None
-    exam_attempt_obj.completed_at = None
-    exam_attempt_obj.allowed_time_limit_mins = None
-    exam_attempt_obj.save()
+    # resetting a submitted attempt that has not been reviewed will entirely remove that submission.
+    if exam_attempt_obj.status == ProctoredExamStudentAttemptStatus.submitted:
+        remove_exam_attempt(exam_attempt_obj.id, requesting_user)
```

Comment:
When this attempt is removed, how do we remove it from Verificient? Is that implemented? If so, is this properly implemented for RPNowV4 exams?

Reply:
there's a signal that calls out to the backend interface to delete. In the case of RPNowV4 this is a no-op (we don't delete attempts with them)

https://github.com/edx/edx-proctoring/blob/master/edx_proctoring/signals.py#L94